### PR TITLE
DB schema generator

### PR DESCRIPTION
**Pull request by cymon1997**
Fix generator version to use, and update guregu null to include v4 (latest)
Add sample migration files 
Generate sample schema generator

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,66 +1,23 @@
 module github.com/cymon1997/go-template
 
-go 1.18
+go 1.16
```
Comment (cymonevolution) [inline go.mod:3]: why do we need to downgrade the go version?

**Review by progrevo — CHANGES_REQUESTED** [review]
Great work!

File: go.mod (hunk 1 above)
Comment (cymon1997) [reply go.mod:3]: for development compatibility because most devs still use 1.16 as their default

Comment (cymonevolution) [reply go.mod:3]: ok then

**Review by progrevo — APPROVED** [review]
LGTM

**Pull request merged**